library/src/blas3/rocblas_trmm.hpp

```diff
@@ -0,0 +1,1488 @@
+                            }
+                            //
+                            //                      T1 := alpha*T1*A + delta*T1, triangular matrix
+                            //                      multiply where the value of delta depends on
+                            //                      whether A is a unit or non-unit triangular
+                            //                      matrix. Gamma and tsec are used to compensate
+                            //                      for a deficiency in DGEMV that appears if the
+                            //                      second dimension (tsec) is zero.
+                            //
+                            for(int i = ii + isec - 1; i >= ii; i--)
+                            {
+                                if(diag == rocblas_diagonal_non_unit)
+                                {
+                                    delta                  = *alpha;
+                                    bool pointer_mode_host = false;
+                                    if(handle->pointer_mode == rocblas_pointer_mode_host)
+                                    {
+                                        pointer_mode_host = true;
+                                        CHECK_INTERNAL_ROCBLAS_ERROR((rocblas_set_pointer_mode(
+                                            handle, rocblas_pointer_mode_device)));
+                                    }
+                                    CHECK_INTERNAL_ROCBLAS_ERROR(
+                                        (rocblas_scal_template<NB, T>(handle,
+                                                                      jsec,
+                                                                      &a[i - 1 + (i - 1) * lda],
+                                                                      0,
+                                                                      &dt1[(i - ii) * ldt1],
+                                                                      0,
+                                                                      1,
+                                                                      0,
+                                                                      1)));
+
+                                    if(pointer_mode_host)
+                                    {
+                                        CHECK_INTERNAL_ROCBLAS_ERROR((rocblas_set_pointer_mode(
+                                            handle, rocblas_pointer_mode_host)));
+                                    }
+                                }
+                                tsec = ii + isec - 1 - i;
+                                tsec = i - ii;
+                                if(0 == tsec)
+                                {
+                                    CHECK_INTERNAL_ROCBLAS_ERROR(
+                                        (rocblas_scal_template<NB, T>(handle,
+                                                                      jsec,
```

Comment:
`<NB, T>` should be able to be replaced with `<NB>`, and then the parentheses around the call are unnecessary.

Reply:
T is required, moved parenthesis to function name and template parameters